chromadb/test/conftest.py

```diff
@@ -133,6 +165,70 @@ def fastapi_persistent() -> Generator[System, None, None]:
     return _fastapi_fixture(is_persistent=True)
 
 
+def fastapi_server_auth() -> Generator[System, None, None]:
+    return _fastapi_fixture(
+        is_persistent=False,
+        chroma_server_auth_provider="chromadb.config.BasicAuthServerProvider",
+        chroma_server_auth_provider_config={"username": "admin", "password": "admin"},
+        chroma_client_auth_provider="chromadb.config.BasicAuthClientProvider",
+        chroma_client_auth_provider_config={"username": "admin", "password": "admin"},
+    )
+
+
+def fastapi_server_auth_param() -> Generator[System, None, None]:
+    for item in _fastapi_fixture(
+        is_persistent=False,
+        chroma_server_auth_provider="chromadb.config.BasicAuthServerProvider",
+        chroma_server_auth_provider_config={"username": "admin", "password": "admin"},
+        chroma_client_auth_provider="chromadb.config.BasicAuthClientProvider",
+        chroma_client_auth_provider_config={"username": "admin", "password": "admin"},
+    ):
+        yield item
+
+
+def fastapi_server_auth_file() -> Generator[System, None, None]:
+    server_auth_file = os.path.abspath(os.path.join(".", "server-auth"))
```

Comment:
lets use a tmpdir (can search the code base for how we do this)

Also does the path have to be absolute? It can be relative too right?